lib/charms/mongodb/v0/mongodb_provider.py

```diff
@@ -79,6 +82,16 @@ def __init__(self, charm: CharmBase, substrate="k8s", relation_name: str = "data
             self.database_provides.on.database_requested, self._on_relation_event
         )
 
+    def _on_relation_departed(self, event):
```

Comment:
Maybe we can put this into charm.py:

Reasoning:
There is no specific action taken in the provider related code. All actions are done in charm.py

Reply:
great feedback! thank you